http-client-spi/src/main/java/software/amazon/awssdk/http/SdkHttpHeaders.java

```diff
@@ -35,4 +37,20 @@
      * @return An unmodifiable map of all headers in this message.
      */
     Map<String, List<String>> headers();
+
+    /**
+     * Perform a case-insensitive search for a particular header in this request, returning the first matching header, if one is
+     * found.
+     *
+     * <p>This is useful for headers like 'Content-Type' or 'Content-Length' of which there is expected to be only one value
+     * present.</p>
+     *
+     * <p>This is equivalent to invoking {@link SdkHttpUtils#firstMatchingHeader(Map, String)}</p>.
+     *
+     * @param header The header to search for (case insensitively).
+     * @return The first header that matched the requested one, or empty if one was not found.
+     */
+    default Optional<String> firstMatchingHeader(String header) {
```

Comment:
If it's in SdkHttpHeaders do you still need to implement it in the Request/Response interfaces?

Reply:
No, but I have to implement it in the builder interfaces.